node_modules/col-assets/tests/test-assets.js

```diff
@@ -40,6 +41,64 @@ var AssetsTestUtil = require('./util');
 describe('Assets', function() {
 
   /**
+   * Get course by Canvas course id
+   *
+   * @param  {Number}           canvasCourseId          The id of the course in Canvas
+   * @param  {Function}         callback                Invoked when the course has been retrieved
+   * @param  {Course}           callback.course         The retrieved course object
+   * @throws {AssertionError}                           Error thrown when an assertion failed
+   */
+  var getCourse = function(canvasCourseId, callback) {
+    var options = {
+      'where': {
+        'canvas_course_id': canvasCourseId
+      },
+      'include': [{
+        'model': DB.Canvas,
+        'as': 'canvas'
+      }]
+    };
+    DB.Course.findOne(options).complete(function(err, course) {
+      assert.ok(!err);
+      assert.ok(course);
+
+      return callback(course);
+    });
+  };
+
+  /**
+   * Get user from database via Canvas user data
+   *
+   * @param  {User}             canvasUser              The user of the course
+   * @param  {Course}           course                  The course in Canvas
+   * @param  {String[]}         sections                The sections of the course in which user is enrolled
+   * @param  {Function}         callback                Invoked when the user has been retrieved
+   * @param  {User}             callback.user           The retrieved user object
+   * @throws {AssertionError}                           Error thrown when an assertion failed
+   */
+  var findUser = function(canvasUser, course, sections, callback) {
+    var options = {
+      'where': {
+        'canvas_user_id': canvasUser.id
+      }
+    };
+    DB.User.findOne(options).complete(function(err, user) {
+      assert.ok(!err);
+
+      var profile = {
+        'canvas_course_role': canvasUser.roles,
+        'canvas_full_name': canvasUser.fullName,
+        'canvas_course_sections': sections
+      };
+      UsersAPI.getOrCreateUser(canvasUser.id, course, profile, function(err, user) {
+        assert.ok(!err);
+
+        callback(user);
+      });
+    });
+  };
```

Comment:
Like `test-poller`, we have direct interaction with db. As far as I know, this is the simplest way to assign sections to the user before verifying search.